src/lib/utils/debounce.js

```diff
@@ -1,15 +1,18 @@
 /**
 	A simple debounce function taken from here https://www.freecodecamp.org/news/javascript-debounce-example/
-	@param {function} func The function to debounce.
+	@param {Function} func The function to debounce.
 	@param {number} timeout The time in ms to wait.
-	@returns {function}
+	@returns {Function}
 */
-export default function debounce(func, timeout = 300) {
+const debounce = (func, timeout = 300) => {
+	/** @type {string | number | NodeJS.Timeout | undefined} */
```

Comment:
This might be more readable (via [Stackoverflow](https://stackoverflow.com/questions/51040703/what-return-type-should-be-used-for-settimeout-in-typescript):

```suggestion
	/** @type {ReturnType<typeof setTimeout>} */
```